lib/fs/basicfs_lstat_windows.go

```diff
@@ -96,19 +93,11 @@ func (f *BasicFilesystem) underlyingLstat(name string) (os.FileInfo, error) {
 	if err == nil {
 		// NTFS directory junctions can be treated as ordinary directories,
 		// see https://forum.syncthing.net/t/option-to-follow-directory-junctions-symbolic-links/14750
-		if fi.Mode()&os.ModeSymlink != 0 && f.junctionsAsDirs {
+		if fi.Mode()&junctionPointModeMask != 0 && f.junctionsAsDirs {
 			if reparseTag, reparseErr := readReparseTag(name); reparseErr == nil && isDirectoryJunction(reparseTag) {
 				return &dirJunctFileInfo{fi}, nil
 			}
 		}
-
-		// Workaround for #9120 till golang properly handles deduplicated files by
-		// considering them regular files.
-		if fi.Mode()&os.ModeIrregular != 0 {
-			if reparseTag, reparseErr := readReparseTag(name); reparseErr == nil && isDeduplicatedFile(reparseTag) {
-				return &dedupFileInfo{fi}, nil
-			}
-		}
```

Comment:
Was this fixed in 1.22 and earlier? I did a search but I don't find the fix for this documented

Reply:
It was fixed in 1.21 in Nov 2023 in
https://github.com/golang/go/commit/f26fa055229502fb6df4b6d9c300eccf01098248